backend/src/libs/plugins/authorization/authorization.plugin.ts

```diff
@@ -0,0 +1,53 @@
+import { type FastifyInstance, type FastifyRequest } from 'fastify';
+import fp from 'fastify-plugin';
+
+import {
+  type HttpMethod,
+  HttpCode,
+  HttpError,
+} from '~/libs/packages/http/http.js';
+
+import { type AuthorizationPluginParameters } from './libs/types/authorization-plugin-parameters.type.js';
+
+const authorization = fp(
+  async (
+    fastify: FastifyInstance,
+    { whiteRoutesConfig, userService, token }: AuthorizationPluginParameters,
+  ) => {
+    fastify.decorateRequest('user', null);
+
+    fastify.addHook('onRequest', async (request: FastifyRequest) => {
+      const isWhiteRoute = whiteRoutesConfig.some((whiteRoute) => {
+        const isWhitePath = whiteRoute.routerPath === request.routerPath;
+        const isAllowedMethod = whiteRoute.methods.includes(
+          request.method as HttpMethod,
+        );
+
+        return isWhitePath && isAllowedMethod;
+      });
+
+      if (isWhiteRoute) {
+        return;
+      }
+
+      const [, requestToken] = request.headers.authorization?.split(' ') ?? [];
+
+      const { userId } = token.decode<{ userId: number }>(requestToken);
```

Comment:
TS does not complain on it, but it actually looks like you can have requestToken equal to `undefined`

Reply:
Would be better to just specify default value or do something like
`const authHeader = request.headers.authorization;
if (!authHeader) {
  throw;
}
const [, requestToken] = authHeader.split(' ');
if (!requestToken) {
  throw
}`

Reply:
Yep, handling by throwing error is nice